lang/java/compiler/src/main/javacc/org/apache/avro/compiler/idl/idl.jj

```diff
@@ -1058,6 +1059,9 @@ private Schema NamedSchemaDeclaration(Map<String, JsonNode> props):
      } else {                                     // add all other props
        Accessor.addProp(s, key, props.get(key));
      }
+   LogicalType logicalType = LogicalTypes.fromSchemaIgnoreInvalid(s);
+   if (logicalType != null)
+     logicalType.addToSchema(s);
```

Comment:
Every time we apply properties to a schema, we should call this logic.
It determines the logical type of the schema, and applies it if possible.